hubblestack/extmods/hubble_mods/win_reg.py

```diff
@@ -0,0 +1,209 @@
+# -*- encoding: utf-8 -*-
+"""
+Module for fetching registry values from windows registry
+
+Audit Example 1:
+---------------
+check_unique_id:
+  description: 'win_reg check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:Microsoft Windows Server 2016*'
+      hubble_version: '>3 AND <7 AND <8'
+      module: win_reg
+      items:
+        - args:
+            name: 'HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention'
+          comparator:
+            type: "dict"
+            match:
+              HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention:
+                 type: "number"
+                 match: "== 0"
+
+Audit Example 2:
+---------------
+check_unique_id:
+  description: 'win_reg check'
+  tag: 'ADOBE-01'
+  implementations:
+    - filter:
+        grains: 'G@osfinger:Microsoft Windows Server 2016*'
+      hubble_version: '>3 AND <7 AND <8'
+      module: win_reg
+      items:
+        - args:
+            name: 'HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft Services\AdmPwd\PasswordAgeDays'
+          comparator:
+            type: "dict"
+            match:
+              HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft Services\AdmPwd\PasswordAgeDays:
+                 type: "number"
+                 match: "<= 30"
+
+FDG Example:
+------------
+main:
+  description: 'win_reg fdg'
+  module: win_reg
+  args:
+    name: HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPreventions
+Mandatory parameters:
+    name - registry name
+
+Note: Comparison logic is moved to comparators. Module will just invoke the win_reg command.
+Comparator compatible with this module - number, string
+
+Sample Output:
+1. can be dictionary with value being an int like
+    {"HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\Explorer\NoDataExecutionPrevention": 0}
+2. can be a dictionary with value being a str like
+    {"HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\NetworkProvider\HardenedPaths\\*\NETLOGON":
```

Comment:
can we have a uniform output of a dict having key as string and integer as value only. The output in this case will become 
{
'RequireMutualAuthentication' : 1,
'RequireIntegrity':1
}

Reply:
I have no control over this. These values are registry values.